packages/datadog-instrumentations/src/child_process.js

```diff
@@ -0,0 +1,176 @@
+'use strict'
+
+const util = require('util')
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const childProcessChannelStart = channel('datadog:child_process:execution:start')
+const childProcessChannelFinish = channel('datadog:child_process:execution:finish')
+const childProcessChannelError = channel('datadog:child_process:execution:error')
+
+// ignored exec method because it calls to execFile directly
+const execAsyncMethods = ['execFile', 'spawn']
+const execSyncMethods = ['execFileSync', 'spawnSync']
+
+const names = ['child_process', 'node:child_process']
+
+// child_process and node:child_process returns the same object instance, we only want to add hooks once
+let patched = false
+names.forEach(name => {
+  addHook({ name }, childProcess => {
+    if (!patched) {
+      patched = true
+      shimmer.massWrap(childProcess, execAsyncMethods, wrapChildProcessAsyncMethod())
+      shimmer.massWrap(childProcess, execSyncMethods, wrapChildProcessSyncMethod())
+      shimmer.wrap(childProcess, 'execSync', wrapChildProcessSyncMethod(true))
+    }
+
+    return childProcess
+  })
+})
+
+function wrapChildProcessSyncMethod (shell = false) {
+  return function wrapMethod (childProcessMethod) {
+    return function () {
+      if (!childProcessChannelStart.hasSubscribers || arguments.length === 0) {
+        return childProcessMethod.apply(this, arguments)
+      }
+
+      const childProcessInfo = {
+        command: arguments[0],
+        args: arguments[1],
+        options: arguments[2]
+      }
+
+      if (Array.isArray(childProcessInfo.args)) {
+        childProcessInfo.command = childProcessInfo.command + ' ' + childProcessInfo.args.join(' ')
```

Comment:
why are we sending, command with arguments from one side, and only the args array from other side?

Reply:
Not sure if I'm following you.The command's argument must be an array, if not it supposed to contain the options. Therefore whenever the second argument is an array its contents are appended to the command if not they are options and not appended to the command